File: easy/array/jewels/jewels.py
class Solution(object):
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        jewels = 0
        for j in J:
            jewels += S.count(j)
        return jewels


s = Solution()
print("Solution 1 : ", s.numJewelsInStones("aA", "aAAbbbb"))
print("Solution 2 : ", s.numJewelsInStones("z", "ZZ"))

File: easy/array/jewels/jewels.go
package main

import "fmt"

func numJewelsInStones(J string, S string) int {
	jewels := make(map[rune]struct{})
	for _, j := range J {
		jewels[j] = struct{}{}
	}

	numOfJewels := 0
	for _, s := range S {
		if _, ok := jewels[s]; ok {
			numOfJewels += 1
		}
	}

	return numOfJewels
}

func main() {
	fmt.Println("Solution 1 : ", numJewelsInStones("aA", "aAAbbbb"))
	fmt.Println("Solution 2 : ", numJewelsInStones("z", "ZZ"))
}
